### ISSUE TITLE
MySQL: Models don't have booleans coerced when using union queries

### ISSUE THREAD

**Issue by sethboyles** [jeremyevans/sequel#1644]
### Complete Description of Issue

When using the `union` function in MySQL (sequel-5.24.0), resulting records do not have their boolean columns coerced from integers to booleans.  This can result in confusing behavior around equality between different instances of the same record:

```
> user = User.union(User.dataset).first
> user == User.find(id: user.id)
=> false
```
This is because doing a simple find results in values like this:
```
> User.find(id: user.id).values
=> {
   :id=>1,
   :active=>true,
}
```
While the union does not coerce the bool columns:
```
> user.values
=> {
    :id=>1,
    :active=>1,
}
```

The equality method compares `values`, so returns false when asking if one user is the same as the other:

```
def eql?(obj)
  (obj.class == model) && (obj.values == @values)
end
``` 

We did not observe this behavior with Postgres.  A record fetched with `union` equals a record fetched normally, as one might expect.  

### Simplest Possible Self-Contained Example Showing the Bug

When having a model with boolean columns:

```
> user = User.union(User.dataset).first
> user == User.find(id: user.id)
=> false
```

**Comment by jeremyevans** [comment]
This is a bug, but it is a bug in MySQL, not a bug in Sequel.  MySQL does not support booleans.  The best you can do is emulate them using `tinyint(1)` (which is what Sequel and other database libraries do).

When using a `UNION`, MySQL drops the `tinyint(1)` size denoting a boolean, and promotes the column to an integer.  MySQL does not even support CASTing the type back to `tinyint`, attempting to do so is a syntax error. See https://dba.stackexchange.com/questions/12569/how-to-cast-an-integer-to-a-boolean-in-a-mysql-select-clause

Basically, I don't think there is a way to work around this, you'll have to fix the bug in MySQL.
